src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1591,119 +1566,184 @@ static size_t delete_monitors(GrowableArray<ObjectMonitor*>* delete_list) {
+    // ObjectMonitors that were deflated in this cycle.
+    HandshakeForDeflation hfd_hc;
+    Handshake::execute(&hfd_hc);
+    // Also, we sync and desync GC threads around the handshake, so that they can
+    // safely read the mark-word and look-through to the object-monitor, without
+    // being afraid that the object-monitor is going away.
+    VM_RendezvousGCThreads sync_gc;
+    VMThread::execute(&sync_gc);
+
+    log.after_handshake();
 
     // After the handshake, safely free the ObjectMonitors that were
     // deflated and unlinked in this cycle.
-    if (current->is_Java_thread()) {
-      if (ls != NULL) {
-        timer.stop();
-        ls->print_cr("before setting blocked: unlinked_count=" SIZE_FORMAT
-                     ", in_use_list stats: ceiling=" SIZE_FORMAT ", count="
-                     SIZE_FORMAT ", max=" SIZE_FORMAT,
-                     unlinked_count, in_use_list_ceiling(),
-                     _in_use_list.count(), _in_use_list.max());
-      }
-      // Mark the calling JavaThread blocked (safepoint safe) while we free
-      // the ObjectMonitors so we don't delay safepoints whilst doing that.
-      ThreadBlockInVM tbivm(JavaThread::cast(current));
-      if (ls != NULL) {
-        ls->print_cr("after setting blocked: in_use_list stats: ceiling="
-                     SIZE_FORMAT ", count=" SIZE_FORMAT ", max=" SIZE_FORMAT,
-                     in_use_list_ceiling(), _in_use_list.count(), _in_use_list.max());
-        timer.start();
-      }
-      deleted_count = delete_monitors(&delete_list);
-      // ThreadBlockInVM is destroyed here
-    } else {
-      // A non-JavaThread can just free the ObjectMonitors:
-      deleted_count = delete_monitors(&delete_list);
-    }
+
+    log.before_blocked(unlinked_count);
+
+    // Mark the calling JavaThread blocked (safepoint safe) while we free
+    // the ObjectMonitors so we don't delay safepoints whilst doing that.
+    ThreadBlockInVM tbivm(current);
+
+    log.after_blocked();
+
+    // Delete the unlinked ObjectMonitors.
+    deleted_count = delete_monitors(&delete_list);
     assert(unlinked_count == deleted_count, "must be");
-  }
 
-  if (ls != nullptr) {
-    timer.stop();
-    if (deflated_count != 0 || unlinked_count != 0 || log_is_enabled(Debug, monitorinflation)) {
-      ls->print_cr("deflated_count=" SIZE_FORMAT ", {unlinked,deleted}_count=" SIZE_FORMAT " monitors in %3.7f secs",
-                   deflated_count, unlinked_count, timer.seconds());
-    }
-    ls->print_cr("end deflating: in_use_list stats: ceiling=" SIZE_FORMAT ", count=" SIZE_FORMAT ", max=" SIZE_FORMAT,
-                 in_use_list_ceiling(), _in_use_list.count(), _in_use_list.max());
+    // ThreadBlockInVM is destroyed here
   }
 
   OM_PERFDATA_OP(MonExtant, set_value(_in_use_list.count()));
   OM_PERFDATA_OP(Deflations, inc(deflated_count));
 
   GVars.stw_random = os::random();
 
+  log.end(deflated_count, unlinked_count);
```

Comment:
Why move this `log.end()` call after the OM_* calls and the os::random() call?

Reply:
I don't remember why I moved it.

Reply:
I moved it back.